cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,229 @@ Cask Data Application Platform Release Notes
+  that the correct container size is used.
+
+- :cask-issue:`CDAP-8913` - Improved the serializability of Tables and IndexedTables when
+  used in Spark programs.
+
+- :cask-issue:`CDAP-8918` - Fixed an issue where redeployment of an application with a
+  deleted schedule would fail.
+
+- :cask-issue:`CDAP-8945` - Moved the "add plugin" behavior from a plugin's left panel to
+  an "Add Entity" button in the CDAP Studio UI.
+
+- :cask-issue:`CDAP-8950` - Fixed an issue in the CDAP UI where navigating from a stream
+  card to an overview and then to a detail page made the detail page show a spinner icon
+  indefinitely.
+
+- :cask-issue:`CDAP-8980` - Fixed an issue with the Spark program runtime so that the Kryo
+  serializer can be used.
+
+- :cask-issue:`CDAP-9005` - Fixed an issue where the HBaseQueueDebugger failed when
+  authorization was enabled.
+
+- :cask-issue:`CDAP-9027` - Added support for grant/revoke on groups, and revoke on users,
+  in the CDAP Sentry Extension.
+
+- :cask-issue:`CDAP-9029` - Fixed an issue where the CDAP Sentry Extension failed to grant
+  privileges to a role.
+
+- :cask-issue:`CDAP-9035` - Fixed an issue where the CDAP Sentry Extension was not able to
+  fetch privileges associated with a user's group.
+
+- :cask-issue:`CDAP-9046` - Fixed an issue where revoking privileges from a role caused
+  the privilege to be revoked from all roles.
+
+- :cask-issue:`CDAP-9084` - LogHandler endpoints now return a 404 status code if the
+  entity (the run id) for which logs are requested does not exist.
+
+- :cask-issue:`CDAP-9086` - Fixed an issue with the Window plugin so that it propagates
+  schema properly.
+
+- :cask-issue:`CDAP-9087` - Fixed the Overview panel in home page of the CDAP UI to handle
+  unknown entities appropriately.
+
+- :cask-issue:`CDAP-9114` - Fixed an issue with the retrying of local dataset operations
+  when a failure happens.
+
+- :cask-issue:`CDAP-9142` - Fixed an issue with the binary format in the Kafka streaming
+  source that prevented pipeline deployment.
+
+- :cask-issue:`CDAP-9160` - Fixed a memory leak issue with the Hadoop FileSystem object.
+
+- :cask-issue:`CDAP-9185` - Fixed an issue with the pipeline upgrade tool that caused it
+  to skip CDAP 4.0.x pipelines.
+
+- :cask-issue:`CDAP-9216` - Fixed bug where navigation links were referencing default
+  namespace instead of the current namespace.
+
+- :cask-issue:`CDAP-9247` - Removed a misleading error log message in the CDAP Explore
+  service.
+
+- :cask-issue:`CDAP-9248` - Fixed bug where the "Total Messages/Errors/Warnings" at the
+  top of logviewer was showing incorrect values.
+
+- :cask-issue:`CDAP-9314` - Made a StructuredRecord serializable by Kryo in a Spark
+  program when the Kryo Serializer is used.
+
+- :cask-issue:`CDAP-9333` - Fixed an issue where preview did not show data for successful
+  stages if a particular stage failed.
+
+- :cask-issue:`HYDRATOR-703` - Improved DB Plugin Get Schema error messaging.
```

Comment:
Improved error messages for the 'Get Schema' functionality of Database plugins in CDAP Pipelines.